builtin/logical/pki/path_acme_account.go

```diff
@@ -230,18 +242,54 @@ func (b *backend) acmeNewAccountCreateHandler(acmeCtx *acmeContext, userCtx *jws
 		return formatAccountResponse(acmeCtx, accountByKey), nil
 	}
 
+	var eab *eabType
+	var eabData map[string]interface{}
+	if acmeCtx.requireEab {
+		eabDataRaw, ok := data["externalAccountBinding"]
```

Comment:
This seems incorrect: if EAB is present, shouldn't we always be verifying it and rejecting it, regardless of `requireEab` status? And only if `requireEab` is assert do we fail if there is no EAB?

Reply:
Fair, addressed in https://github.com/hashicorp/vault/commit/493ca5734ba600ab6f0aaa796a84073a1e9f67f8